packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx

```diff
@@ -173,25 +173,30 @@ export const _TagsSurvey = (props: _TagsSurveyProps): JSX.Element => {
           placeholder={strings?.tagsSurveyTextFieldDefaultText}
           onChange={(e, v) => {
             if (v) {
+              setCheckedTextFields([...checkedTextFields, issueCategory]);
               setTextResponse((prevState) => {
                 prevState[issueCategory] = v;
                 return prevState;
               });
 
               setSelectedTextResponse((prevState) => {
-                if (Object.keys(prevState).includes(issueCategory)) {
-                  prevState[issueCategory] = v;
-                }
+                prevState[issueCategory] = v;
                 return prevState;
               });
             }
           }}
         />
       );
     },
-    [strings?.tagsSurveyTextFieldDefaultText]
+    [strings?.tagsSurveyTextFieldDefaultText, checkedTextFields]
   );
 
+  useEffect(() => {
+    if (onConfirm) {
```

Comment:
this onConfirm is no longer happening in an onChange event. This will execute on the first render when nothing has changed which feel incorrect

Reply:
When is onConfirm supposed to fire? This will also fire anytime selectedTextResponse or selectedTags changes

Reply:
Onconfirm is used for the tagSurvey composite components to update the state variables that stores the selected tags info. So it needs to fire anytime selectedTextResponse or selectedTags changes